x/programs/rust/examples/single_nft/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+    program
+        .state()
+        .store(StateKey::TotalSupply.to_vec(), &TOTAL_SUPPLY)
+        .expect("failed to store total supply");
+
+    // Initialize counter
+    program
+        .state()
+        .store(StateKey::Counter.to_vec(), &0_i64)
+        .expect("failed to store counter");
+
+    true
+}
+
+/// Mints NFT tokens and sends them to the recipient.
+#[public]
+pub fn mint(program: Program, recipient: Address) -> bool {
+    const MINT_AMOUNT: i64 = 1;
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .expect("failed to store ");
+
+    assert!(counter <= TOTAL_SUPPLY, "max supply for nft exceeded");
+
+    let balance = program
+        .state()
+        .get::<i64, _>(StateKey::Balance(recipient).to_vec())
+        .unwrap_or_default();
+
+    program
+        .state()
+        .store(
+            StateKey::Balance(recipient).to_vec(),
+            &(balance + MINT_AMOUNT),
+        )
+        .expect("failed to store balance");
+
+    program
+        .state()
+        .store(StateKey::Owner.to_vec(), &recipient)
+        .expect("failed to store owner");
+
+    program
+        .state()
+        .store(StateKey::Counter.to_vec(), &(counter + 1_i64))
+        .is_ok()
+}
+
+#[public]
+pub fn burn(program: Program, from: Address) -> bool {
+    const BURN_AMOUNT: i64 = 1;
+    let null_address = Address::new([0; 32]);
+
+    // Only the owner of the NFT can burn it
+    let owner = program
+        .state()
+        .get::<Address, _>(StateKey::Owner.to_vec())
+        .unwrap_or(null_address);
+
+    assert_eq!(owner, from, "only the owner can burn the nft");
+
+    let balance = program
+        .state()
+        .get::<i64, _>(StateKey::Balance(from).to_vec())
+        .unwrap_or_default();
+
+    assert!(
+        BURN_AMOUNT <= balance,
+        "amount burned must be less than or equal to the user balance"
+    );
+
+    let counter = program
+        .state()
+        .get::<i64, _>(StateKey::Counter.to_vec())
+        .unwrap_or_default();
+
+    assert!(counter > 0, "cannot burn more nfts");
+
+    // Burn the NFT by transferring it to the zero address
+    program
+        .state()
+        .store(StateKey::Balance(from).to_vec(), &(balance - BURN_AMOUNT))
+        .expect("failed to store new balance");
+
+    // TODO move to a lazy static? Or move to the VM layer entirely
+    program
+        .state()
+        .store(StateKey::Owner.to_vec(), &null_address)
+        .is_ok()
+}
+
+#[cfg(test)]
+mod tests {
+    use serial_test::serial;
+    use std::env;
+    use wasmlanche_sdk::simulator::{self, Key, PlanResponse};
+
+    use crate::example;
+
+    // export SIMULATOR_PATH=/path/to/simulator
+    // export PROGRAM_PATH=/path/to/program.wasm
+    // cargo cargo test --package token --lib nocapture -- tests::test_token_plan --exact --nocapture --ignored
+    #[test]
+    #[serial]
+    #[ignore = "requires SIMULATOR_PATH and PROGRAM_PATH to be set"]
```

Comment:
I really don't like that we're using `ignore` like this, is there no better way to do this?

Reply:
Please reply to my comments before resolving them

Reply:
This is based on @hexfusion's work in token. Do you have an alternate suggestion?

Reply:
Yeah, I left another comment a while ago. I suggested that you use the `env!` macro instead so that when you go to run the tests, they won't even compile if you don't specify the paths.

Reply:
I'm not sure exactly what you're referring to here. Could you show me?

Reply:
https://github.com/ava-labs/hypersdk/pull/587

Reply:
^^ the above PR is how you would do it... but it looks like there's more work to be done here that can happen after this PR lands.